src/main/java/io/crate/jmx/recorder/Recorder.java

```diff
@@ -45,6 +45,14 @@ default boolean recordBean(String domain,
                                                 " cannot be called with CompositeData bean value");
     }
 
+    default boolean recordBean(String domain,
+                               String attrName,
+                               CompositeData[] beanValue,
+                               MetricSampleConsumer metricSampleConsumer) {
+        throw new UnsupportedOperationException(this.getClass().getSimpleName() +
+                " cannot be called with CompositeData[] bean value");
+    }
+
```

Comment:
I wonder if it would make sense to simplify the `Recorder` Interface to a single method of the signature:

```
    default boolean recordBean(String domain,
                               String attrName,
                               Object beanValue,
                               MetricSampleConsumer metricSampleConsumer) {
```

remove the other methods and handle the type specific logic inside the actual `Recorder` implementations.